### PR TITLE
Re-implement classic quarried stone gen

### PR DESCRIPTION
**Pull request by andreblanke**
**The Issue**

Railcraft Reborn currently generates quarried stone using the regular `minecraft:ore` feature. I've always liked the more unique ways Railcraft used to generate resources and would thus like to see it reintroduced.
 
**The Proposal**

This PR restores the classic quarried stone generation using a custom `railcraft:quarried` feature. The old generation logic has mostly been ported over and is made a bit more flexible through the use of `QuarryConfiguration` as configured feature, allowing configuration of the blocks to place (i.e., `railcraft:quarried_stone` by default) and the target blocks which are allowed to be replaced during generation (i.e., all blocks matched by the newly introduced `#railcraft:quarried_replaceable_blocks` tag).

**Possible Side Effects**

I'm not aware of any side effects except changing the generated amount of quarried stone back to its original amount which is likely lower than the amount currently generated using the `minecraft:ore` feature.
 
**Alternatives**

The alternative would be to keep the status quo.

### CODE REVIEW

**Comment by Edivad99** [comment]
Run `./gradlew spotlessApply` to fix the formatting

File: src/main/java/mods/railcraft/world/level/levelgen/feature/RailcraftFeatures.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import net.neoforged.bus.api.IEventBus;
+import net.neoforged.neoforge.registries.DeferredHolder;
+import net.neoforged.neoforge.registries.DeferredRegister;
+
+import mods.railcraft.api.core.RailcraftConstants;
+import mods.railcraft.world.level.levelgen.feature.configuration.QuarriedConfiguration;
+
+public class RailcraftFeatures {
+
+    private static final DeferredRegister<Feature<?>> FEATURES =
```
Comment (Edivad99) [inline src/main/java/mods/railcraft/world/level/levelgen/feature/RailcraftFeatures.java]: I know it's right to put `FEATURES`, but better to put `deferredRegister` for consistency with the rest. One day I'll change the name of all of them.
```suggestion
    private static final DeferredRegister<Feature<?>> deferredRegister =
```

File: src/main/java/mods/railcraft/world/level/levelgen/feature/RailcraftFeatures.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        DeferredRegister.create(Registries.FEATURE, RailcraftConstants.ID);
+
+    public static final DeferredHolder<Feature<?>, QuarriedFeature> QUARRIED_STONE =
+        FEATURES.register("quarried", () -> new QuarriedFeature(QuarriedConfiguration.CODEC));
```
Comment (Edivad99) [inline src/main/java/mods/railcraft/world/level/levelgen/feature/RailcraftFeatures.java]:
```suggestion
        deferredRegister.register("quarried", () -> new QuarriedFeature(QuarriedConfiguration.CODEC));
```

File: src/main/java/mods/railcraft/world/level/levelgen/feature/RailcraftFeatures.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    public static void register(IEventBus eventBus) {
+        FEATURES.register(eventBus);
```
Comment (Edivad99) [inline src/main/java/mods/railcraft/world/level/levelgen/feature/RailcraftFeatures.java]:
```suggestion
        deferredRegister.register(eventBus);
```

**Review by Edivad99 — CHANGES_REQUESTED** [review]

**Review by Edivad99 — APPROVED** [review]

**Comment by andreblanke** [comment]
Sorry for that, I was just about to push the changes you requested

**Comment by Edivad99** [comment]
Oops, I wanted to test it and ended up fixing a few things😅

**Comment by Edivad99** [comment]
Well done, by the way👏🏻

**Comment by andreblanke** [comment]
> Well done, by the way👏🏻

Thank you, also for resolving the remaining formatting issues and merging it. I might look into the sulfur world generation as well.

**Comment by Edivad99** [comment]
Ok, so I'm waiting to do the release for 1.21. I was thinking of doing it today, but now I'll wait.

**Comment by andreblanke** [comment]
I might be able to finish it today, but I could use some input regarding the generation details. I've started the discussion #231 for it.

**Pull request merged**